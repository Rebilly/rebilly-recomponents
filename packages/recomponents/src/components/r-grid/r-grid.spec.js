import {mount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RGrid from './r-grid.vue';

const provider = async () => [
    {id: 1, name: 'One', type: 'Odd'},
    {id: 2, name: 'Two', type: ['Even', 'Prime']},
    {
        id: 3, name: 'Three', type: ['Odd', 'Prime'], ignore: 'IGNORED',
    },
];

const columns = {
    columns: [
        {
            name: 'id',
            renderAs: 'numeric',
        },
        {
            name: 'name',
            style: {
                minWidth: '300px',
                textAlign: 'center',
            },
            renderAs: 'text',
        },
        {
            name: 'type',
            renderAs: 'badge',
            renderOptions: {
                type: 'negative',
            },
        },
    ],
};

describe('r-grid.vue', () => {
    it('should render Wrapper and match snapshot', () => {
        const wrapper = mount(RGrid, {
            propsData: {provider, columns},
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', () => {
        const wrapper = renderToString(RGrid, {
            propsData: {provider, columns},
        });

        expect(wrapper).toMatchSnapshot();
    });
});
