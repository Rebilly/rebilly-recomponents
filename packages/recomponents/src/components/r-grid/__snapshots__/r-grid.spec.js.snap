// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-grid.vue should render Wrapper and match snapshot 1`] = `
<div class="r-grid">
  <div class="r-grid-body">
    <div class="r-table-control flex-center r-grid-repeater">
      <div role="alert" aria-busy="true" class="r-component r-loader r-is-loading r-is-fullscreen">
        <div class="r-loader-icon-wrapper">
          <div class="r-mr-re"></div>
          <div class="r-mr-bil"></div>
          <div class="r-mr-ly"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`r-grid.vue should render via SSR and match snapshot 1`] = `
<div class="r-grid">
  <div class="r-grid-body">
    <div class="r-table-control flex-center r-grid-repeater">
      <div role="alert" aria-busy="true" class="r-component r-loader r-is-loading r-is-fullscreen">
        <div class="r-loader-icon-wrapper">
          <div class="r-mr-re"></div>
          <div class="r-mr-bil"></div>
          <div class="r-mr-ly"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
